src/nvim/api/buffer.c

```diff
@@ -1172,14 +1192,10 @@ void nvim_buf_delete(Buffer buffer, Dictionary opts, Error *err)
     return;
   }
 
-  int result = do_buffer(unload ? DOBUF_UNLOAD : DOBUF_WIPE,
-                         DOBUF_FIRST,
-                         FORWARD,
-                         buf->handle,
-                         force);
+  int result = do_buffer(action, DOBUF_FIRST, FORWARD, buf->handle, force);
 
   if (result == FAIL) {
-    api_set_error(err, kErrorTypeException, "Failed to unload buffer.");
+    api_set_error(err, kErrorTypeException, "Failed to delete buffer.");
```

Comment:
It would be nice if we could store the action string and use that instead of "delete". `api_set_error` already supports format strings so it should be trivial to do so.